.github/workflows/self-test.yml

```diff
@@ -18,6 +18,9 @@ on:
 
 jobs:
   test:
+    permissions:
+      issues: write
+      pull-requests: write
```

Comment:
I think Github changed the token's default permissions recently. These additional lines solved it, but another user reported the same problem in https://github.com/cpp-linter/cpp-linter/issues/89#issuecomment-1999641987

We may have to update the README and examples recipes about this.

Reply:
I cherry-picked the relevant commits to a new branch and submitted the changes in #225

When #225 is merged, I'll drop the commits to this branch and rebase this branch.